lib/fabric/app/models/fabric/event.rb

```diff
@@ -20,37 +21,27 @@ class Event
     index({ stripe_id: 1 }, { background: true, unique: true })
     index({ webhook: 1, customer_id: 1 }, background: true)
 
-    # stripe_id and webhook cannot be obtained from event.data.object, so
-    # events must be initialized with them before running #from
     def sync_with(event)
-      self.api_version = event[:api_version]
-      self.created = event[:created]
-      self.data = event[:data]
-      self.livemode = event[:livemode]
-      self.request = event[:request]
-      customer_id = event[:object] == 'customer' ? event[:id] : event[:customer]
-      self.customer_id = customer_id
+      self.api_version = event.api_version
+      self.created = event.created
+      event_data = handle_hash(event.data)
+      self.data = event_data if Fabric.config.store_events_data
+      self.livemode = event.livemode
+      self.request = handle_hash(event.request)
+      self.customer_id = Fabric::Event.extract_customer_id(event_data)
       self
     end
 
-    def stripe_object_id
-      data.try(:[], :object).try(:[], :id)
-    end
-
-    # get stripe customer id from data.object
-    # TODO: test if not tested
-    def stripe_customer
-      if is_customer?
-        data[:object].try(:[], :id)
-      else # object == 'subscription', etc.
-        data[:object].try(:[], :customer)
-      end
-    end
-
-    private
-
-    def is_customer?
-      data[:object].try(:[], :object) == 'customer'
+    # this actually is the stupid structure of event['data']
+    # {
+    #   object: {
+    #     object: 'customer/setup_intent/payment_intent.etc',
+    #     customer_id: 'cus_xxx'
+    #   }
+    # }
+    def self.extract_customer_id(event_data)
```

Comment:
It never worked because we don't actually use `sync_with` when we create events. This method is never actually used, but exists because every model has one. `Fabric::Webhook#create_event` is where the events get created. I checked production data and none of the events have the `customer_id` set. Events are kinda odd.